src/User.php

```diff
@@ -174,6 +174,11 @@ public function canDeleteItem()
             return false;
         }
 
+        //prevent delete / purge from API
+        if (!self::isMassiveActionAllowed($this->fields['id'])) {
+            return false;
+        }
```

Comment:
Maybe the whole PR can be replaced by
```suggestion
        global $CFG_GLPI;
        if ($this->fields['id'] == $CFG_GLPI['system_user']) {
            return false;
        }
```

Reply:
It may confuse the users, I think @stonebuzz solution where the checkbox is disabled is clearer.

Reply:
This part can be effectively simplified.

But is only useful in the case of a deletion from the

Reply:
Last commit use direct check instead of call ```isMassiveActionAllowed```